src/main/java/com/minecolonies/tileentities/ScarecrowTileEntity.java

```diff
@@ -0,0 +1,161 @@
+package com.minecolonies.tileentities;
+
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.ColonyManager;
+import com.minecolonies.inventory.InventoryField;
+import com.minecolonies.util.EntityUtils;
+import com.minecolonies.util.LanguageHandler;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntityChest;
+import net.minecraft.world.World;
+
+import java.util.Random;
+
+/**
+ * The scarecrow tile entity to store extra data.
+ */
+public class ScarecrowTileEntity extends TileEntityChest
+{
+    /**
+     * NBTTag to store the type.
+     */
+    private static final String TAG_TYPE = "type";
+
+    /**
+     * NBTag to store the name.
+     */
+    private static final String TAG_NAME = "name";
+
+    /**
+     * Random generator.
+     */
+    private final Random random          = new Random();
+
+    /**
+     * The inventory connected with the scarecrow.
+     */
+    private InventoryField inventoryField;
+
+    /**
+     * The type of the scarecrow.
+     */
+    private ScareCrowType  type;
+
+    /**
+     * Name of the scarecrow, string set in the GUI.
+     */
+    private String name = LanguageHandler.format("com.minecolonies.gui.scarecrow.user", LanguageHandler.format("com.minecolonies.gui.scarecrow.user.noone"));
+
+    /**
+     * Creates an instance of the tileEntity.
+     */
+    public ScarecrowTileEntity()
+    {
+        super();
+        this.inventoryField = new InventoryField(name);
+    }
+
+    /**
+     * Getter of the name of the tileEntity.
+     * @return the string.
+     */
+    public String getName()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1161)